src/mpyl/reporting/formatting/markdown.py

```diff
@@ -124,47 +128,43 @@ def to_markdown_test_report(suites: list[TestSuite]):
     return f"{summary_to_markdown(total_tests)}"
 
 
-def _collect_test_artifacts(step_results: list[StepResult]) -> list[Artifact]:
-    return [
-        res.output.produced_artifact
+def _collect_test_artifacts(step_results: list[StepResult]) -> dict[str, JunitTestSpec]:
```

Comment:
we should maybe rename the function as well, although technically it's still a test artifact anyway I guess

Reply:
https://github.com/Vandebron/mpyl/pull/185/commits/980d7053973d46fa83a93bc5aad4ccdc4f364371